pydantic/fields.py

```diff
@@ -367,16 +384,16 @@ class MyModel(pydantic.BaseModel):
         if annotation is default:
             raise PydanticUserError(
                 'Error when building FieldInfo from annotated attribute. '
-                "Make sure you don't have any field name clashing with a type annotation ",
+                "Make sure you don't have any field name clashing with a type annotation.",
                 code='unevaluable-type-annotation',
             )
 
         final = _typing_extra.is_finalvar(annotation)
-        if final and annotation is not typing_extensions.Final:
+        if final and _typing_extra.is_generic_alias(annotation):
```

Comment:
Curious - why do you feel like it's better to use the generic alias check here, rather than just `is not typing_extensions.Final`? Maybe we could add another comment here explaining that it's a parametrized `Final`, just so that it's clear to future devs...

Reply:
At some point, `typing_extensions` might not reexport `typing.<member>` when an upstream cpython change is backported in `typing_extensions`. This means that `typing_extensions.Final != typing.Final` can be `True` in the future.

See https://typing-extensions.readthedocs.io/en/latest/#runtime-use-of-types.

> Maybe we could add another comment here explaining that it's a parametrized `Final`, just so that it's clear to future devs...

The logic here is meant to be merged with the one from `from_annotation`, but leaving it for https://github.com/pydantic/pydantic/issues/11122.